From c8bd7997020f413d3e9da11d77c177b66ac22866 Mon Sep 17 00:00:00 2001
From: "Verma, Aashish" <aashishx.verma@intel.com>
Date: Thu, 25 Jul 2019 01:06:19 +0800
Subject: [PATCH 03/52] net: stmmac: Fix missing IFF_MULTICAST check in
 dwmac4_set_filter

Without checking for IFF_MULTICAST flag, it is wrong to assume multicast
filtering is always enabled. As a result, we cannot disable multicast
filtering in MAC.

Fixes: 477286b53f55 stmmac: add GMAC4 core support

Signed-off-by: Verma, Aashish <aashishx.verma@intel.com>
Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c b/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c
index 3751675..9f9a596 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c
@@ -426,7 +426,7 @@ static void dwmac4_set_filter(struct mac_device_info *hw,
 		/* Set all the bits of the HASH tab */
 		for (i = 0; i < numhashregs; i++)
 			writel(0xffffffff, ioaddr + GMAC_HASH_TAB(i));
-	} else if (!netdev_mc_empty(dev)) {
+	} else if (!netdev_mc_empty(dev) && (dev->flags & IFF_MULTICAST)) {
 		struct netdev_hw_addr *ha;
 		u32 mc_filter[8];
 
-- 
2.7.4

